src/main/java/com/spring/onlinebookstore/repository/cart/ShoppingCartRepository.java

```diff
@@ -0,0 +1,9 @@
+package com.spring.onlinebookstore.repository.cart;
+
+import com.spring.onlinebookstore.model.ShoppingCart;
+import java.util.Optional;
+import org.springframework.data.jpa.repository.JpaRepository;
+
+public interface ShoppingCartRepository extends JpaRepository<ShoppingCart, Long> {
+    Optional<ShoppingCart> findByUserId(Long id);
```

Comment:
Comment related to `@Fetch` annotation in `ShoppingCart` entity. Here, for example, I don't think you need/want to load User relation with all it's data and have an extra join, but please check it according to your logic